src/UI/Button.elm

```diff
@@ -1,30 +1,73 @@
 module UI.Button exposing
-    ( Button
-    , ButtonBody
-    , ButtonStyle
-    , ButtonWidth
-    , clear
-    , cmd
-    , danger
-    , disabled
-    , fromIcon
-    , fromLabel
-    , full
-    , hyperlink
-    , light
-    , map
-    , primary
-    , redirect
-    , renderElement
-    , shrink
-    , success
-    , toggle
-    , withDisabledIf
+    ( Button, toggle, success, disabled, cmd, redirect
+    , ButtonBody, fromLabel, fromIcon
+    , ButtonStyle, hyperlink, primary, danger, light, clear
+    , ButtonWidth, withWidth, full, shrink
     , withSize
-    , withSuccessIf
-    , withWidth
+    , withDisabledIf, withSuccessIf
+    , renderElement
+    , map
     )
 
+{-| The button is the unified aggregation of all the standardized clickable elements.
+The same component can render as a hyperlink, an embossed button, or a padded-icon.
+
+The embossed-styled button has four tones: Primary, Danger, Light, Clear.
+And it can also be in one of the states: Active, Disabled, and Success.
```

Comment:
Non-pr related, maybe we should call `Enabled` instead of `Active` as we are using `Disabled` and not `Deactivated`